dump/plugins/fdb.py

```diff
@@ -0,0 +1,90 @@
+from dump.match_infra import MatchRequest
+from dump.helper import create_template_dict
+from .executor import Executor
+
+
+class Fdb(Executor):
+    """
+    Debug Dump Plugin for FDB Module
+    """
+    ARG_NAME = "Vlan:fdb_entry"
+
+    def __init__(self, match_engine=None):
+        super().__init__(match_engine)
+
+    def get_all_args(self, ns=""):
+        req = MatchRequest(db="STATE_DB", table="FDB_TABLE", key_pattern="*", ns=ns)
+        ret = self.match_engine.fetch(req)
+        fdb_entries = ret["keys"]
+        return [key.split("|")[-1] for key in fdb_entries]
+
+    def execute(self, params):
+        self.ret_temp = create_template_dict(dbs=["APPL_DB", "ASIC_DB", "STATE_DB"])
+        fdb_entry = params[Fdb.ARG_NAME]
+        self.ns = params["namespace"]
+        self.init_fdb_appl_info(fdb_entry)
+        self.init_asic_fdb_info(fdb_entry)
+        self.init_state_fdb_info(fdb_entry)
+        return self.ret_temp
+
+    def init_state_fdb_info(self, fdb_name):
+        req = MatchRequest(db="STATE_DB", table="FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+
+    def init_fdb_appl_info(self, fdb_name):
+        req = MatchRequest(db="APPL_DB", table="FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"], False)
+        req = MatchRequest(db="APPL_DB", table="VXLAN_FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"], False)
+        req = MatchRequest(db="APPL_DB", table="MCLAG_FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"], False)
+
+    def init_asic_fdb_info(self, fdb_name):
+        key_split = fdb_name.split(":",1)
```

Comment:
Will be good to check if ":" exists in fdb_name before splitting and update ret_temp accordingly.

Reply:
Done